### PR TITLE
Move serverVulnerabilityAssessment API to stable-2020-01-01

### PR DESCRIPTION
**Pull request by aaviram**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/serverVulnerabilityAssessments.json
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/serverVulnerabilityAssessments.json:336]: what value does this resource type add when it just gives back the provisioning state property to the user?

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/serverVulnerabilityAssessments.json
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/serverVulnerabilityAssessments.json:37]: do you support this for only the top level resource types in an RP? So, as per current modeling this will not work for databases resource type inside MIcrosoft.SQL. since for databases the resource id is /subscriptions/{id}/resourceGroups/{rg}/providers/Microsoft.SQL/servers/{name}/databases/{dbName}/
There is no way to put /databases/{name} in the URI

**Review by ravbhatnagar — CHANGES_REQUESTED** [review]
@aaviram - few comments

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/serverVulnerabilityAssessments.json
Comment (aaviram) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/serverVulnerabilityAssessments.json:336]: This provisions a vulnerability assessment scanner on a VM. once it is installed, the user can have scans and see results in Azure Security Center Assessments API.

Just noting that this API was already added ~year ago for preview- This is just a copy in a Stable API.

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/serverVulnerabilityAssessments.json
Comment (aaviram) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/serverVulnerabilityAssessments.json:37]: This is solely for IaaS VMs

**Comment by aaviram** [comment]
@ravbhatnagar 
Can anyone give me a hint why these tests are failing? Locally this passes

**Comment by ArcturusZhang** [comment]
> @ravbhatnagar
> Can anyone give me a hint why these tests are failing? Locally this passes

Hi @aaviram 
The Avocado CI failure is complaining that the new swagger file is not referenced by the readme. You may need to add your new swagger to a new tag or an existing tag in `readme.md`.
The ModelValidation CI failure is saying that some of your parameters in examples do not follow their particular patterns, but it seems like a false alarm.
Hi @raych1 would you please have a look why the Model Validation is failing? The subscription ID in the example should be valid.

And @aaviram please take some time fix the Avocado first, maybe after the Avocado got fixed, Model Validation would fix itself

**Comment by raych1** [comment]
@ArcturusZhang , model validation has a bug in pattern match and was fixed in staging validation check.

**Comment by aaviram** [comment]
Hey @ravbhatnagar , Still waiting for this PR to be approved and completed, can anyone please have a look?

**Comment by ArcturusZhang** [comment]
@aaviram  Also could you please have a look at the incompatible values error? It is blocking the generation of SDKs, for instance:
```
2020-04-20T05:10:39.132Z: [Azure/azure-sdk-for-go][AutoRest] Process() cancelled due to exception : '$.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceNamespace}/{resourceType}/{resourceName}/providers/Microsoft.Security/serverVulnerabilityAssessments"].get["x-ms-examples"]["Get a list of server vulnerability assessments on a resource. Though this API returns a list, Currently Microsoft.Security only supports a single default type of server vulnerability assessment"].parameters["api-version"]' has incompatible values (---
2020-04-20T05:10:39.132Z: [Azure/azure-sdk-for-go][AutoRest] 2019-01-01-preview
2020-04-20T05:10:39.132Z: [Azure/azure-sdk-for-go][AutoRest] , ---
2020-04-20T05:10:39.132Z: [Azure/azure-sdk-for-go][AutoRest] '2020-01-01'
2020-04-20T05:10:39.132Z: [Azure/azure-sdk-for-go][AutoRest] ).
```

**Comment by aaviram** [comment]
@ArcturusZhang - All validations passed- Can someone please merge the changes?

Thanks

**Comment by aaviram** [comment]
@ravbhatnagar - Can someone please merge this?

**Comment by ArcturusZhang** [comment]
@ravbhatnagar could you please have a look at the new changes?

@aaviram could you please resolve the merge conflict?

File: specification/security/resource-manager/readme.md
Comment (ArcturusZhang) [inline specification/security/resource-manager/readme.md:1]: This file contains unresolved merge conflicts, you may need to recheck this

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
Please recheck this PR to ensure there is no more unresolved merge conflicts

File: specification/security/resource-manager/readme.md
Comment (aaviram) [reply specification/security/resource-manager/readme.md]: Resolved.
Can someone please merge?

Thanks

**Comment by ArcturusZhang** [comment]
Hi @ravbhatnagar could you please have a look at this PR? the author has made some changes.

**Comment by aaviram** [comment]
@ravbhatnagar , @ArcturusZhang @raych1   This PR is waiting from Apr16, And it will be now the second time that I need to solve conflicts because no one gets back to me. Why doesn't anybody help to make this over?

**Comment by ArcturusZhang** [comment]
Hi @aaviram I understand the frustration and I am very sorry about this. I have wrote an email to ARM, hopefully someone will come, have a look and sign off.

**Comment by ArcturusZhang** [comment]
Now an issue introduced in PR #8235 is causing all SDK generation to fail...

Problematic swagger: https://github.com/Azure/azure-rest-api-specs/blob/7eeef3ce97572bf15bbf3865fdfd17e78b1e760b/specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/secureScore.json#L534
In the `enum` attribute, there is only `BuiltIn` defined, but in the following `x-ms-enum` property, there are two values defined.

@chlahav @giromm2ms  Could you please have a look?

**Comment by aaviram** [comment]
@ArcturusZhang - I can't understand- How come something that isn't affected by this PR breaks validation?...... Why does this PR still need to wait again and again?

**Comment by aaviram** [comment]
@Addys

**Comment by ArcturusZhang** [comment]
> @ArcturusZhang - I can't understand- How come something that isn't affected by this PR breaks validation?...... Why does this PR still need to wait again and again?

We have to wait for the ARM to sign off before we can merge this PR... And I have written an email to ARM (should with your cc), let me ping them again.
And the new failure was introduced in another merged PR #8235  , when you merge master back, the error was introduced to this PR...
@giromm2ms @allenjzhang could you please help verify the error introduced by #8235 ?

**Comment by giromm2ms** [comment]
Error fixed and merged couple of days ago.
Please rebase from today since PR 9583 added some other changes.

Thanks and sorry for the inconvenience.

**Comment by ArcturusZhang** [comment]
Thanks @giromm2ms !

And now the only blocker should be ARM's approval. I just pinged them in another email, hopefully someone could come here and take a look at this...

**Review by KrisBash — APPROVED** [review]
I believe Gaurav's comments were addressed.

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**
